apps/lockfile-explorer-web/src/parsing/compareSpec.ts

```diff
@@ -0,0 +1,57 @@
+import { IPackageJson } from '../types/IPackageJson';
+
+export interface ISpecChange {
+  type: string;
+  pkg: string;
+  from?: string;
+  to?: string;
+}
+
+export const compareSpec = (
```

Comment:
The `src/parsing/*.ts` modules are central parts of your business logic.  Each of these functions should have a `/** */` comment. It doesn't need to be anything elaborate, but should cover the basics of:

- The big picture of what problem we're solving (what it's for, not how it works)
- An idea of how it's used by the app
- Some concrete examples of its inputs and outputs where appropriate

Reply:
For example:
```ts
/**
 * Calculate the diff between a package.json file and its transformed "package spec".
 * 
 * @remarks
 * During installation, PNPM applies various transforms to `package.json` files, for example 
 * .pnpmfile.cjs may add/remove/rewrite version ranges. The transformed result is called 
 * the "package spec" by Lockfile Explorer, and its tab pane displays the diff between 
 * the original `package.json` and the final spec. 
 * 
 * @returns A map of `ISpecChange` differences, looked up by package name. For example: 
 *
 * 'react' --> { packageName: 'react', type: 'diff', from: '^16.0.0', to: '~16.2.0' }
 */
```

Reply:
Added